### DISCUSSION TITLE
predictableActionArguments: `meta` migration

### DISCUSSION THREAD

**Discussion by LekoArts** [statelyai/xstate#3511] [Q&A] [answered]
Hi!

Bringing my question in https://github.com/statelyai/xstate/pull/3289#discussion_r942162140 to here.

So at Gatsby we use (and love) xstate, specifically in the `gatsby` and `gatsby-source-fileystem` package. We recently saw the warning message in the `gatsby develop` output and wanted to migrate to remove that warning. A WIP PR is here: https://github.com/gatsbyjs/gatsby/pull/36342

On my YOLO update I missed this note:

> Please be aware that you might not able to use `state` from the `meta` argument when using this flag.

And our unit tests failed, because we use that pattern in `gatsby-source-filesystem`. Here's the commit fixing it: [gatsbyjs/gatsby@`efb5d12` (#36342)](https://github.com/gatsbyjs/gatsby/pull/36342/commits/efb5d12320eb66fa82f5b0e81cda49fc6ba3d46e)

You can [view the file](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-filesystem/src/gatsby-node.js) to see the whole state machine and see the context where this is used (used to `log` things to the console). It's like a "conditional action", so only log to the console if the state is `BOOTSTRAP.BOOTSTRAPPED`.

That's basically our use case. Are there better ways to migrate/Would this use case warrant a note on the docs?

Thanks!

**Comment by Andarist** [comment, marked as answer]
I've created a PR to your PR that hopefully will allow you to migrate quickly: https://github.com/gatsbyjs/gatsby/pull/36355 😉
